src/Compilers/VisualBasic/Portable/Symbols/Wrapped/WrappedMethodSymbol.vb

```diff
@@ -0,0 +1,264 @@
+﻿' Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+Imports System.Collections.Immutable
+Imports System.Globalization
+Imports System.Reflection
+Imports System.Threading
+Imports Microsoft.Cci
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
+    ''' <summary>
+    ''' Represents a method that is based on another method.
+    ''' When inheriting from this class, one shouldn't assume that 
+    ''' the default behavior it has is appropriate for every case.
+    ''' That behavior should be carefully reviewed and derived type
+    ''' should override behavior as appropriate.
+    ''' </summary>
+    Friend MustInherit Class WrappedMethodSymbol
+        Inherits MethodSymbol
+
+        Public MustOverride ReadOnly Property UnderlyingMethod() As MethodSymbol
```

Comment:
In C# WrappedXXXX types are also used in other non-tuple symbols. It is a convenient pattern for all symbols that do some kind of wrapping. That is not yet done in VB since it is not required for tuples, but I would like to keep code similar between VB and C# here to make it easier to port those changes

---
In reply to: [71573455](https://github.com/dotnet/roslyn/pull/12596#discussion_r71573455) [](ancestors = 71573455)